spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/core/subscriber/PubSubSubscriberTemplateTests.java

```diff
@@ -530,6 +531,21 @@ void testPullAndConvertAsync()
     assertThat(result.get(0).getProjectSubscriptionName().getSubscription()).isEqualTo("sub2");
   }
 
+  @Test
+  void testPullAndConvertAsync_publishesConvertException()
+          throws InterruptedException, ExecutionException, TimeoutException {
+    when(this.messageConverter.fromPubSubMessage(this.pubsubMessage, BigInteger.class)).thenThrow(new NullPointerException());
+    CompletableFuture<List<ConvertedAcknowledgeablePubsubMessage<BigInteger>>> asyncResult =
+            this.pubSubSubscriberTemplate.pullAndConvertAsync("sub2", 1, true, BigInteger.class);
+
+    try {
+      asyncResult.get();
+      Assertions.fail();
```

Comment:
Add a message that appears when this assertion fails.

Reply:
done